src/foundry/common/utils/helpers.mjs.d.ts

```diff
@@ -275,7 +306,7 @@ type MergeObjectProperty<T, U, M extends MergeObjectOptions> = T extends Array<a
   ? U extends Record<string, any>
     ? M extends { recursive: false }
       ? U
-      : Result<T, U, Omit<M, 'insertKeys'> & { insertKeys: M['insertValues'] }>
+      : Result<T, U, Omit<M, 'insertKeys' | 'performDeletions'> & { insertKeys: M['insertValues'] }>
```

Comment:
does it actually behave correctly with respect to `performDeletions`? I.e., is that covered in the type? We should add some tests for that.

Reply:
No it does not but I've yet to try to unwrap the type, shoving `'performDeletions'` here does silence the issues with adding that key to the options.

Reply:
I'll give this explicit thing more of a shake tomorrow.